src/Generator/ShoppingCOM.php

```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
+
+            $this->addCSVContent(array_values($data));
+
+            $this->getLogger(__METHOD__)->debug('ElasticExportShoppingCOM::log.variationConstructRowFinished', [
+                'Data row duration' => 'Row printing took: ' . (microtime(true) - $rowTime),
+            ]);
+        }
+        else
+        {
+            $this->getLogger(__METHOD__)->info('ElasticExportShoppingCOM::log.variationNotPartOfExportPrice', [
+                'VariationId' => (string)$variation['id']
+            ]);
+        }
+    }
+
+    /**
+     * Get the shipping cost.
+     *
+     * @param $variation
+     * @param $settings
+     * @return string
+     */
+    private function getShippingCost($variation, $settings):string
+    {
+        $shippingCost = $this->elasticExportHelper->getShippingCost($variation['data']['item']['id'], $settings, 0);
+
+        if(!is_null($shippingCost) && $shippingCost > 0)
+        {
+            return number_format((float)$shippingCost, 2, ',', '');
+        }
+
+        return '';
+    }
+
+    /**
+     * Get the manufacturer name.
+     *
+     * @param $variation
+     * @return string
+     */
+    private function getManufacturer($variation):string
+    {
+        if(isset($this->manufacturerCache) && array_key_exists($variation['data']['item']['manufacturer']['id'], $this->manufacturerCache))
+        {
+            return $this->manufacturerCache[$variation['data']['item']['manufacturer']['id']];
+        }
+
+        return '';
     }
 
     /**
-     * @param RecordList $idlResultList
+     * Build the cache arrays for the item variation.
+     *
+     * @param $variation
      */
-    private function createIdlArray($idlResultList)
+    private function buildCaches($variation)
     {
-        if($idlResultList instanceof RecordList)
+        if(!is_null($variation) && !is_null($variation['data']['item']['id']))
         {
-            foreach($idlResultList as $idlVariation)
+            if(!is_null($variation['data']['item']['manufacturer']['id']))
             {
-                if($idlVariation instanceof Record)
+                if(!isset($this->manufacturerCache) || (isset($this->manufacturerCache) && !array_key_exists($variation['data']['item']['manufacturer']['id'], $this->manufacturerCache)))
                 {
-                    $this->idlVariations[$idlVariation->variationBase->id] = [
-                        'itemBase.id' => $idlVariation->itemBase->id,
-                        'variationBase.id' => $idlVariation->variationBase->id,
-                        'itemPropertyList' => $idlVariation->itemPropertyList,
-                        'variationRetailPrice.price' => $idlVariation->variationRetailPrice->price,
-                        'variationRetailPrice.vatValue' => $idlVariation->variationRetailPrice->vatValue,
-                    ];
+                    $manufacturer = $this->elasticExportHelper->getExternalManufacturerName((int)$variation['data']['item']['manufacturer']['id']);
```

Comment:
we do not need a cache for manufacturer because we already cache it at the eeHelper